lib/resty/upstream/healthcheck.lua

```diff
@@ -661,22 +649,105 @@ local function gen_peers_status_info(peers, bits, idx)
         if peer.down then
             bits[idx + 2] = " DOWN\n"
         else
-            bits[idx + 2] = " up\n"
+            bits[idx + 2] = " UP\n"
         end
         idx = idx + 3
     end
     return idx
 end
 
+local new_status_meta = {
+    __add = function(self, rhs)
+        -- debug("new_status_meta:__add: rhs: ", rhs)
+        self.statuses[self.idx] = rhs
+        self.idx = self.idx + 1
+    end
+}
+new_status_meta.__index = new_status_meta
+
+function new_status_meta:add(rhs)
+    self:__add(rhs)
+end
+
+local function new_status_table(n)
+    local tab = {
+        statuses = new_tab(n * 20, 0),
+        idx = 1
+    }
+    return setmetatable(tab, new_status_meta)
+end
+
+-- upstream status generator function
+
+local function gen_upstream_prometheus_status(u, s, n)
```

Comment:
If we add the string directly to the table without formatting， there would fewer GC strings.
When there are lots of upstream servers, it is very important.

Remember to add the "\n" yourself, and concat without the "\n"